.github/workflows/workflow.yml

```diff
@@ -76,10 +76,52 @@ jobs:
           ~/test-cache
     - name: Verify cache files in working directory
       shell: bash
-      run: __tests__/verify-cache-files.sh ${{ runner.os }} test-cache
+      run: __tests__/verify-cache-files.sh ${{ runner.os }}-save test-cache
     - name: Verify cache files outside working directory
       shell: bash
-      run: __tests__/verify-cache-files.sh ${{ runner.os }} ~/test-cache
+      run: __tests__/verify-cache-files.sh ${{ runner.os }}-save ~/test-cache
+
+  # End to end with save-always
+  test-save-always:
```

Comment:
```suggestion
  # End to end with save-always
  test-save-always-with-failure:
```

What do you think about changing the name here to make it clear that these jobs are expected to fail?

I want to make it clear that the three failing checks are expected

![PR Merge box with three failing checks from this job](https://github.com/actions/cache/assets/16631042/3567679d-3f3d-43ad-b377-eb6eea3054c9)

Reply:
Is there any way to do something like a post hook to check that failure occurred and mark as passing? It does seem prone to confusion to have CI marked intentionally as failing

Reply:
Yep, completely agree.

Reply:
Yeah I'm a little unclear on how this will ultimately look through the github UI. The status of the workflow in the Actions tab is is passing, but the indicator above for checks on the commit show failing. Maybe that's technically accurate, but still feels a little weird. Either way that's a bigger issue than this bug or this repo.

I don't _think_ there is a way to force the step conclusions to pass without altering the post step to get a `success`, but perhaps Josh knows more.

Reply:
Agreed it's confusing - I don't think we support anyway to hide these. If you wanted a step to fail but not fail the job itself, you can add `continue-on-error` to the step but that wouldn't allow us to test the `post-if` case where `success()` is false.